### PR TITLE
Add extract to SD card feature

### PR DESCRIPTION
**Pull request by Shailin**
|PR checklist|Status|
|-----|:------:|
|Is issue created in GitHub?|[Yes](https://github.com/maqsoftware/GLEXP-Team-CCI-Hindi-2/issues/19)||
|Is code analysis task executed and issues resolved in the Java/Kotlin code?|Yes|
|Is code analysis task executed and issues resolved in the CPP/native code?|NA|
|Is the version code set in the AndroidManifest.xml or build.gradle file to 1 more than the version which has already been published?|NA|
|Is code commenting done properly for the changes made?|Yes|
|Is correct variable naming convention followed?|Yes|
|Is auto formatting shortcut Ctrl+Alt+Shift+L executed in all files?|Yes|
|Are the message strings used in the application taken from strings.xml file?|Yes|
|Are the permissions which are not required removed from AndroidManifest.java file?|Yes|
|Is the _uses-feature_ tag used for only must have feature in AndroidManifest.java file?|NA|
|Are the generated files correctly configured in .gitignore?|Yes|
|Are the release signing configurations removed from the gradle files?||
|Is commented code removed from all files?|Yes|
|Are updated images compressed (< 100 KB)?|NA|
|Ensured there are no typos in any string. For example: Variable/method names, hardcoded strings, commit messages, PR titles, issues, code comments?|Yes|

### CODE REVIEW

File: platforms/android/app/src/main/java/com/maq/xprize/cci/hindi/SplashScreenActivity.java [hunk 1]
```diff
@@ -1,219 +1,326 @@
...
+            decorView.setSystemUiVisibility(uiOptions);
+        }
+        setContentView(R.layout.activity_splash_screen);
+
+        // if SD card available show Dialog to ask for user preference
+        if (isSDcard() && mainFileVersion == 0) {
+            flagSwitchToInternal = true;
+            Dialog builder = sdCardPreferenceDialog();
+            builder.show();
+        } else if (isSDcard() && sharedPref.getInt("dataPath", 0) == 2) {
```
Comment (ankitk20) [inline platforms/android/app/src/main/java/com/maq/xprize/cci/hindi/SplashScreenActivity.java]: Add comment for this scenario

File: platforms/android/app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -3,9 +3,14 @@
...
     <string name="launcher_name">@string/app_name</string>
     <string name="activity_name">@string/launcher_name</string>
-    <string name="content_extraction">यह प्रक्रिया केवल एक बार की जाएगी। इसमें १० मिनट तक का समय लग सकता है।</string>
+    <string name="content_extraction_hindi">यह प्रक्रिया केवल एक बार की जाएगी। इसमें १० मिनट तक का समय लग सकता है।</string>
+    <string name="content_extraction_english">This process will be done only once. It can take up to 10 minutes.</string>
     <string name="splash_screen">Splash Screen</string>
     <string name="startPercent">0 %</string>
     <string name="mainFileVersion">mainFileVersion</string>
     <string name="patchFileVersion">patchFileVersion</string>
+    <string name="dataPath">dataPath</string>
```
Comment (ankitk20) [inline platforms/android/app/src/main/res/values/strings.xml]: You have used \n to separate in English and Hindi in SD card but have used separate strings in for extraction screen. Can you make it consistent with \n?

Comment (Shailin) [reply platforms/android/app/src/main/res/values/strings.xml]: Using different textView in the extraction screen gives more flexibility for positioning. With '\n' the vertical spacing will not be achieved.

File: platforms/android/app/src/main/java/com/maq/xprize/cci/hindi/SplashScreenActivity.java (hunk 1 above)
Comment (Shailin) [reply platforms/android/app/src/main/java/com/maq/xprize/cci/hindi/SplashScreenActivity.java]: Added comments

File: platforms/android/app/src/main/java/com/maq/xprize/cci/hindi/SplashScreenActivity.java [hunk 3]
```diff
@@ -1,219 +1,327 @@
...
+            toCallApplication();
+        } catch (IOException e) {
+            System.out.println(e);
+        }
+    }
+
+    public boolean isStorageSpaceAvailable() {
+        long totalExpansionFileSize = 0;
+        // check the storage space of the asset path selected
+        File internalStorageDir = new File(assetsPath);
```
Comment (akashdabhi03) [inline platforms/android/app/src/main/java/com/maq/xprize/cci/hindi/SplashScreenActivity.java]: Rename to storageDir to avoid confusion

**Pull request merged**